xapian-letor/ranker/ranker.cc

```diff
@@ -293,8 +294,16 @@ Xapian::prepare_training_file(const string & db_path, const string & queryfile,
 					     parser.FLAG_DEFAULT|
 					     parser.FLAG_SPELLING_CORRECTION,
 					     "S");
+	// query with 'description' field as prefix "XD"
+	Xapian::Query query_description_prefix = parser.parse_query(querystr,
+					parser.FLAG_SPELLING_CORRECTION,
+					"XD");
 	// Combine queries
-	Xapian::Query query = Xapian::Query(Xapian::Query::OP_OR, query_no_prefix, query_default_prefix);
+	Xapian::Query query = Xapian::Query(Xapian::Query::OP_OR,
+				query_no_prefix,
+				Xapian::Query(Xapian::Query::OP_OR,
+					      query_default_prefix,
+					      query_description_prefix));
```

Comment:
Like elsewhere, we seem to be pointlessly specifying `FLAG_SPELLING_CORRECTION` here.

You're not specifying `FLAG_DEFAULT` when parsing with `"XD"` but we are for the other cases, which I guess is not deliberate?

Also it seems what we actually ought to be doing here is setting the default prefix for the query parser to map to "", "S" and "XD", then parsing the query string just once.  Something like:

```
parser.add_prefix("", "");
parser.add_prefix("", "S");
parser.add_prefix("", "XD");
Xapian::Query query = parser.parse_query(querystr);
```

Reply:
fixed in a fixup commit.